packages/plugins/opentelemetry/src/spans.ts

```diff
@@ -311,18 +372,20 @@ export function createSubgraphExecuteSpan(input: {
   executionRequest: ExecutionRequest;
   subgraphName: string;
 }): Context {
+  const operation = getOperationASTFromDocument(
+    input.executionRequest.document,
+    input.executionRequest.operationName,
+  );
+
   const span = input.tracer.startSpan(
     `subgraph.execute (${input.subgraphName})`,
     {
       attributes: {
-        [SEMATTRS_GRAPHQL_OPERATION_NAME]: input.executionRequest.operationName,
+        [SEMATTRS_GRAPHQL_OPERATION_NAME]: operation.name?.value ?? 'Anonymous',
```

Comment:
Since the following is possible, it seems like an empty string indicating the absence would be a better fit.

```
query Anonymous {
  __typename
}
```

Reply:
Yes you are right. I kept it for backward compatibility at first, but now that we decided to cut a v2, I will change that. You prefer an empty string over the absence of the attribute ? It's the default behavior for optional attributes, the are just not reported if the value is undefined.